src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -382,11 +539,8 @@ public bool Disconnect(DisconnectParams disconnectParams)
                 }
             }
 
-            // Close the connection            
-            info.SqlConnection.Close();
-
-            // Remove URI mapping
-            ownerToConnectionMap.Remove(disconnectParams.OwnerUri);
+            // TODO do these activities depend on the type of connection?
```

Comment:
Wondering what others think of this. Should `InvokeOnConnectionActivities` be run after every call to `Connect()`? Likewise for `InvokeOnDisconnectionActivities` and `Disconnect()`.